### ISSUE TITLE
Unknown word error when attempting to parse valid CSS

### ISSUE THREAD

**Issue by mbaumbach** [postcss/postcss-custom-properties#261]
This was found and [detailed in another project](https://github.com/Shopify/polaris-react/issues/4838), but I suspect this is a bug with this project (or perhaps one of its dependencies). The following line:

```css
.Polaris-RangeSlider-SingleThumb__Output { position: absolute; z-index: 20; bottom: var(--p-range-slider-thumb-size-base); left: var(--Polaris-RangeSlider-progress); transform: translateX(calc(-50% + var(--Polaris-RangeSlider-output-factor)*var(--p-range-slider-thumb-size-base))); opacity: 0; visibility: hidden; pointer-events: none; transition-property: opacity, visibility, bottom; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.25, 0.1, 0.25, 1); }
```

Throws the following error:

```
Syntax error: postcss-custom-properties: <css input> Unknown word (1:1)
> 1 | -50% + var(--Polaris-RangeSlider-output-factor)*var(--p-range-slider-thumb-size-base)
    | ^
```

When I manually modify the CSS to include a space around the asterisk between the two custom properties, everything compiles correctly. This code looks like valid CSS to me though, so I think this is a bug in the parser. This code does correctly parse:

```css
.Polaris-RangeSlider-SingleThumb__Output { position: absolute; z-index: 20; bottom: var(--p-range-slider-thumb-size-base); left: var(--Polaris-RangeSlider-progress); transform: translateX(calc(-50% + var(--Polaris-RangeSlider-output-factor) * var(--p-range-slider-thumb-size-base))); opacity: 0; visibility: hidden; pointer-events: none; transition-property: opacity, visibility, bottom; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.25, 0.1, 0.25, 1); }
```

The link above provides steps to reproduce the error. This is part of a create-react-app 5.0.0 project using the Shopify Polaris 7.5.0 framework.

**Comment by mbaumbach** [comment]
And just to confirm, the pulled in `postcss-custom-properties` dependency is v12.0.0.

**Comment by mbaumbach** [comment]
I think this comes down to a failure to parse when the operator is directly preceding the `var`. For example, if you add this test case to the `basic.css` (Which is a slight variation on the `.text--calc` test case):

```css
.test--calc-no-spaces {
	width: calc((100% - var(--xxx, 1px)) +var(--yyy, 10px));
}
```

The `basic.expect.css` theoretically should be:

```css
.test--calc-no-spaces {
	width: calc((100% - 1px) +10px);
	width: calc((100% - var(--xxx, 1px))+var(--yyy, 10px));
}
```

However, the result is as follows where the second `var` wasn't replaced:

```css
	width: calc((100% - 1px) +var(--yyy, 10px));
	width: calc((100% - var(--xxx, 1px))+var(--yyy, 10px));
```

If you change the test to have a space between the `+` and `var`:

```css
.test--calc-no-spaces {
	width: calc((100% - var(--xxx, 1px))+ var(--yyy, 10px));
}
```

The result is correct:

```css
.test--calc-no-spaces {
	width: calc((100% - 1px)+ 10px);
	width: calc((100% - var(--xxx, 1px))+ var(--yyy, 10px));
}
```

I ran these tests against the latest commit.

**Comment by mbaumbach** [comment]
This appears to be fixed in the latest commits, perhaps with the [change to the new parser](https://github.com/postcss/postcss-custom-properties/pull/260). I created a sample test in `basic.css`:

```css
.test--calc-no-spaces {
	transform: translateX(calc(-50% + var(--xxx, 1)*var(--yyy, 10px)));
}
```

That resulted in:

```css
.test--calc-no-spaces {
	transform: translateX(calc(-50% + 1*10px));
	transform: translateX(calc(-50% + var(--xxx, 1)*var(--yyy, 10px)));
}
```

Is there any way to get a new release out to NPM with the new parser in place or is there more work to do there?

**Comment by mbaumbach** [comment]
Confirmed this is fixed in 12.0.1.
